library/hls/src/main/java/com/google/android/exoplayer2/source/hls/HlsChunkSource.java

```diff
@@ -270,29 +270,34 @@ public long getAdjustedSeekPositionUs(long positionUs, SeekParameters seekParame
       return positionUs;
     }
 
+    long startOfPlaylistInPeriodUs =
+        mediaPlaylist.startTimeUs - playlistTracker.getInitialStartTimeUs();
+    long targetPositionInPlaylistUs = positionUs - startOfPlaylistInPeriodUs;
+
+    if (targetPositionInPlaylistUs > mediaPlaylist.durationUs) {
+      return positionUs;
+    }
+
     // Segments start with sync samples (i.e., EXT-X-INDEPENDENT-SEGMENTS is set) and the playlist
-    // is non-empty, so we can use segment start times as sync points. Note that in the rare case
-    // that (a) an adaptive quality switch occurs between the adjustment and the seek being
+    // is non-empty, and the playlist is fresh enough that the period position falls in side of
+    // the playlist bound, so we can use segment start times as sync points.
+    //
+    // If (a) an adaptive quality switch occurs between the adjustment and the seek being
     // performed, and (b) segment start times are not aligned across variants, it's possible that
```

Comment:
Add comment about position exceeds duration check, note track switch on seek is not really a rare condition.  And the stale playlist issue happens for live if there are lots of ABR changes.